src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -406,48 +406,36 @@ def process_file_upload_batch_parameters(namespace):
     # 2. try to extract account name and container name from destination string
     from .storage_url_helpers import StorageResourceIdentifier
     identifier = StorageResourceIdentifier(namespace.destination)
-    if not identifier.is_url():
-        namespace.destination_share_name = namespace.destination
-    elif identifier.filename or identifier.directory:
-        raise ValueError('incorrect usage: destination must be a file share url')
-    else:
-        namespace.destination_share_name = identifier.share
+    if identifier.is_url():
+        if identifier.filename or identifier.directory:
+            raise ValueError('incorrect usage: destination must be a file share url')
 
-    if not namespace.account_name:
-        namespace.account_name = identifier.account_name
+        namespace.destination = identifier.share
 
-    # 3. collect the files to upload
-    namespace.source = os.path.realpath(namespace.source)
-    namespace.source_files = [c for c in glob_files_locally(namespace.source, namespace.pattern)]
+        if not namespace.account_name:
+            namespace.account_name = identifier.account_name
 
-    # 4. clean up
-    del namespace.destination
-    del namespace.source
-    del namespace.pattern
+    namespace.source = os.path.realpath(namespace.source)
 
 
 def process_file_download_batch_parameters(namespace):
     """Process the parameters for storage file batch download command"""
 
     # 1. quick check
     if not os.path.exists(namespace.destination) or not os.path.isdir(namespace.destination):
-        raise ValueError('incorrect usage: --destination must be an existing directory')
+        raise ValueError('incorrect usage: destination must be an existing directory')
 
     # 2. try to extract account name and share name from source string
     from .storage_url_helpers import StorageResourceIdentifier
     identifier = StorageResourceIdentifier(namespace.source)
-    if not identifier.is_url():
-        namespace.source_share_name = namespace.source
-    elif identifier.filename or identifier.directory:
-        raise ValueError('incorrect usage: --source should be either share URL or name')
-    else:
-        namespace.source_share_name = identifier.share
+    if identifier.is_url():
+        if identifier.filename or identifier.directory:
+            raise ValueError('incorrect usage: source should be either share URL or name')
```

Comment:
Similar logic as above. If I don't want to recursively download the entire share, but only a directory within a share, I should be able to do that.